experimental/packages/opentelemetry-sdk-metrics-base/src/utils.ts

```diff
@@ -34,7 +34,7 @@ export function hashAttributes(attributes: MetricAttributes): string {
   keys = keys.sort();
   return keys.reduce((result, key) => {
     if (result.length > 2) {
-      result += ',';
+      result += '|#';
```

Comment:
Are you sure it is safe to use this as a separator if we're allowing string values? I think it is possible to collide hashes, but do we think it is sufficiently unlikely?

Reply:
For string values, it is always possible to collide before or after this change, like:

- `key1` = `value1`, and `key2` = `value2`, results in `|#key1:value1,key2:value2`,
- `key1` = `value1,key2=value2`, also results in `|#key1:value1,key2:value2`.

The change here is mainly to circumvent the string value of an array of strings, like `["value1", "value2"].toString()` is `value1,value2` in which values are separated with `","`.

Reply:
For what is in my mind now, it can be costly to do so. This hashing is very common across the operations in metrics SDK. I would be open to any improvements to this function but I'd be hesitant to make it a relatively costly one.

Reply:
@dyladan I've updated the hash function to support arbitrary attribute values with `JSON.stringify`. PTAL :)